hyperledger/iroha
# refactor: remove committed_topology from block

## Description

- committed topology is removed from block
- committed topology for latest and previous blocks are stored in the state
    - it's required for blocks sync
    - and for initializing state from the snapshot + block store
- initial topology in the genesis block is passed as transaction with `Register<Peer>` instructions
- trusted peers are removed from `World` constructor





### Linked issue



Closes #4879 



### Benefits

Less space used in the block store, easier to derive block hash.



### How to test

Run, try to submit new blocks, try restart peers check that iroha operates as usual.
